### PR TITLE
BXMSPROD-685 upgrade of different dependencies /hibernate, xmlschema

### PR DESCRIPTION
**Pull request by mareknovotny**
Excluding for now Weld upgrade and keeping other changes from #1320

Merge together with:

- https://github.com/kiegroup/appformer/pull/982
- https://github.com/kiegroup/drools/pull/2932
- https://github.com/kiegroup/jbpm/pull/1669
- https://github.com/kiegroup/droolsjbpm-integration/pull/2127
- https://github.com/kiegroup/jbpm-work-items/pull/132
- https://github.com/kiegroup/optaweb-employee-rostering/pull/450

### CODE REVIEW

**Comment by mareknovotny** [comment]
Jenkins execute full downstream build

File: pom.xml [hunk 1]
```diff
@@ -179,9 +179,9 @@
     <version.org.glassfish.jaxb>2.3.2</version.org.glassfish.jaxb>
     <version.org.gwtbootstrap3>1.0.1</version.org.gwtbootstrap3>
     <version.org.gwtbootstrap3-extras>1.0.2</version.org.gwtbootstrap3-extras>
-    <version.org.hibernate>5.3.7.Final</version.org.hibernate>
+    <version.org.hibernate>5.3.15.Final</version.org.hibernate>
```
Comment (gmunozfe) [inline pom.xml:182]: For Oracle 19, we need at least 5.3.16, due to a bug to resolve dialect:
https://hibernate.atlassian.net/browse/HHH-13184
Fixed at 5.3.16
https://hibernate.atlassian.net/secure/ReleaseNote.jspa?version=31822&styleName=Html&projectId=10031

**Comment by mareknovotny** [comment]
Jenkins execute full downstream build

**Comment by mareknovotny** [comment]
Jenkins execute full downstream build

**Comment by mareknovotny** [comment]
Jenkins execute full downstream build

**Comment by mareknovotny** [comment]
Jenkins execute full downstream build

**Comment by mareknovotny** [comment]
Knonw problem in `org.kie.hacep.LocalStorageStreamingKieSessionTest.fireUntilHaltTest`  https://issues.redhat.com/browse/DROOLS-5321

```
java.util.NoSuchElementException
	at java.util.ArrayList$Itr.next(ArrayList.java:862)
	at org.kie.hacep.LocalStorageStreamingKieSessionTest.fireUntilHaltTest(LocalStorageStreamingKieSessionTest.java:91)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

we are ready to merge this. I reran the CDB on appformer kiegroup/appformer#982
and drools kiegroup/drools#2932 just for sake of having green status on their CDB

**Comment by mareknovotny** [comment]
Jenkins execute full downstream build

**Pull request merged**